lib/rules/key-spacing.js

```diff
@@ -72,6 +72,36 @@ function isSingleLine(node) {
     return (node.loc.end.line === node.loc.start.line);
 }
 
+/** Sets option values from the configured options with defaults
+ * @param {Object} toOptions Object to be initialized
+ * @param {Object} fromOptions Object to be initialized from
+ * @param {Object} defultOptions Object to be used for default values
+ * @returns {Object} The object with correctly initialized options and values
+ */
+function initOptions(toOptions, fromOptions, defultOptions) {
+    toOptions.mode = fromOptions.mode || defultOptions.mode || "strict";
+
+    // Set value of beforeColon
+    if (typeof fromOptions.beforeColon !== "undefined") {
+        toOptions.beforeColon = +fromOptions.beforeColon;
+    } else if (typeof defultOptions.beforeColon !== "undefined") {
+        toOptions.beforeColon = +defultOptions.beforeColon;
+    } else {
+        toOptions.beforeColon = 0;
+    }
+
+    // Set value of afterColon
+    if (typeof fromOptions.afterColon !== "undefined") {
+        toOptions.afterColon = +!(fromOptions.afterColon === false);
```

Comment:
One last tweak: `+!(xyz === false)` is equivalent to just `+xyz` in this case since we know `afterColon` will always be `true` or `false` here.